lib/bolero/src/test/mod.rs

```diff
@@ -90,7 +95,7 @@ impl TestEngine {
             .map(move |seed| input::RngTest { seed, max_len })
     }
 
-    fn tests(&self) -> Vec<NamedTest> {
+    fn tests(&self) -> impl Iterator<Item = NamedTest> {
```

Comment:
Hmm I guess this should be a nice performance improvement, especially when tests fail on the first few iterations.